File: Coding_Test/JavaScript/Algorithm/Programmers/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    let progress = []
    for (p of progresses) {
        p = 100 - p
        progress.push(p)
    }

    while (progress.length) {
        let days = 0
        if (progress[0] % speeds[0] == 0) {
            days = parseInt(progress[0] / speeds[0])
        } else {
            days = parseInt(progress[0] / speeds[0] + 1)
        }

        for (let i = 0; i < progress.length; i++) {
            progress[i] -= days * speeds[i]
        }
        let count = 0
        let i = 0
        while (true) {

            if (progress[i] <= 0) {
                count += 1
                progress.shift()
                speeds.shift()
            } else {
                break
            }
        }
        answer.push(count)


    }
    return answer;

}

File: Coding_Test/Python/Algorithm/Programmers/기능개발.py
import copy
from collections import deque


def solution(progresses, speeds):
    arr = []
    for i in range(len(progresses)):
        if (100 - progresses[i]) % speeds[i] == 0:
            arr.append(int((100 - progresses[i]) / speeds[i]))
        else:
            arr.append(int((100 - progresses[i]) / speeds[i]) + 1)

    copyarr = copy.deepcopy(arr)
    copyarr = deque(copyarr)
    count = 1
    max = 0
    result = []
    print(arr)
    while copyarr:
        m = copyarr.popleft()
        if m > max:
            result.append(count)
            max = m
            count = 1
        else:
            count += 1

    result.append(count)

    return result[1:]